geth/common/types.go

```diff
@@ -173,28 +173,11 @@ type TxQueueManager interface {
 	DiscardTransactions(ids string) map[string]RawDiscardTransactionResult
 }
 
-// JailExecutor defines an interface which exposes method to be executed
-// against a Jail vm.
-type JailExecutor interface {
-	// Run exist so we are able to execute js code on pure otto.VM without runing
-	// it on the event loop.
-	Run(string) (otto.Value, error)
-
-	// Exec exists for the purpose to execute has normal on the event loop provided by
-	// ottoext.
-	Exec(string) (otto.Value, error)
-
-	// Fetch calls the underlying FetchAPI which makes http request
-	// to desired path. (See https://developer.mozilla.org/en/docs/Web/API/Fetch_API).
-	Fetch(string, func(otto.Value)) (otto.Value, error)
-}
-
 // JailCell represents single jail cell, which is basically a JavaScript VM.
 type JailCell interface {
-	CellVM() *otto.Otto
-	CellLoop() *loop.Loop
-	Executor() JailExecutor
-	Copy() (JailCell, error)
+	VM() *otto.Otto
```

Comment:
@tiabc But cases like this https://github.com/status-im/status-go/blob/77d01c424e5dab818ce07cff38bdd9779ecdc2c8/geth/api/backend_jail_test.go#L615-L622, would make that impossible, without first doing a type asset for the `JailManager` interface.